File: app/src/main/java/com/summer/itis/summerproject/model/db_dop_models/UserRelation.java
package com.summer.itis.summerproject.model.db_dop_models;

import com.google.firebase.database.IgnoreExtraProperties;

import java.util.HashMap;
import java.util.Map;

@IgnoreExtraProperties
public class UserRelation extends ElementId {

    private String relation;

    private static final String FIELD_RELATION = "relation";

    public UserRelation() {
    }

    public String getRelation() {
        return relation;
    }

    public void setRelation(String relation) {
        this.relation = relation;
    }

    public static Map<String, Object> toMap(String id, String relation) {
        HashMap<String, Object> result = new HashMap<>();
        result.put(FIELD_ID, id);
        result.put(FIELD_RELATION, relation);
        return result;
    }
}

File: app/src/main/java/com/summer/itis/summerproject/model/db_dop_models/UserRelation.kt
package com.summer.itis.summerproject.model.db_dop_models

import com.google.firebase.database.IgnoreExtraProperties

import java.util.HashMap

//ОТНОШЕНИЕ МЕЖДУ ЮЗЕРАМИ(ДРУЗЬЯ И Т.Д)
@IgnoreExtraProperties
class UserRelation : ElementId() {

    var relation: String? = null

    companion object {

        private val FIELD_RELATION = "relation"

        protected val FIELD_ID = "id"

        fun toMap(id: String, relation: String): Map<String, Any> {
            val result = HashMap<String, Any>()
            result[FIELD_ID] = id
            result[FIELD_RELATION] = relation
            return result
        }
    }
}
